commands/secret_create.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) OpenFaaS Author(s) 2019. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package commands
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/openfaas/faas-cli/proxy"
+	"github.com/spf13/cobra"
+)
+
+var (
+	secretName  string
+	secretValue string
+	secretFile  string
+)
+
+// secretCreateCmd represents the secretCreate command
+var secretCreateCmd = &cobra.Command{
+	Use:   "create SECRET_NAME [--from-literal=SECRET_VALUE] [--from-file=/path/to/secret/file] [STDIN]",
+	Short: "Create a new secret",
+	Long:  `The create command creates a new secret from file, literal or STDIN`,
+	Example: `faas-cli secret create secret-name --from-literal=secret-value
+faas-cli secret create secret-name --from-literal=secret-value --gateway=http://127.0.0.1:8080
+faas-cli secret create secret-name --from-file=/path/to/secret/file --gateway=http://127.0.0.1:8080
+cat /path/to/secret/file | faas-cli secret create secret-name`,
+	RunE:    runSecretCreate,
+	PreRunE: preRunSecretCreate,
+}
+
+func init() {
+	secretCreateCmd.Flags().StringVar(&secretValue, "from-literal", "", "Value of the secret")
+	secretCreateCmd.Flags().StringVar(&secretFile, "from-file", "", "Path to the secret file")
+	secretCreateCmd.Flags().BoolVar(&tlsInsecure, "tls-no-verify", false, "Disable TLS validation")
+	secretCreateCmd.Flags().StringVarP(&gateway, "gateway", "g", defaultGateway, "Gateway URL starting with http(s)://")
+	secretCmd.AddCommand(secretCreateCmd)
+}
+
+func preRunSecretCreate(cmd *cobra.Command, args []string) error {
+
+	return nil
+}
+
+func runSecretCreate(cmd *cobra.Command, args []string) error {
+	var gatewayAddress string
+	var err error
+
+	if len(args) < 1 {
+		return fmt.Errorf("please provide secret name")
+	}
+	secretName = args[0]
+
+	if len(secretValue) == 0 && len(secretFile) > 0 {
+		secretValue, err = readSecretFromFile(secretFile)
+		if err != nil {
+			return err
+		}
+
+	}
+
+	if len(secretValue) == 0 && len(secretFile) == 0 {
```

Comment:
What will happen if someone passes both from literal and file?
This is possible with the flags, but we don’t cover the use-case.

Reply:
This seems like a good place for a `switch` (i would also suggest changing the name fo the variable for the `from-literal` flag to something like `literalSecret`)

```golang
var secretValue string
switch {
case len(literalSecret) > 0:
    // handle input
case len(secretFile) > 0:
   // handle parsing from the file
default:
   // handle parsing from stdin
}
```

This will set a clear order of precedence and also make it clear that reading directly from stdin is the default behavior

Reply:
Seems fair as a comment. Can it be done as a refactoring exercise?

Reply:
Yeah, that can be done. I had already left a comment what should be preference between different flags ?